src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5716,12 +5716,22 @@ private bool IsOpenName()
                             explicitInterfaceName,
                             AddError(separator, ErrorCode.ERR_ExplicitEventFieldImpl));
 
-                        Debug.Assert(beforeIdentifierPointSet);
-                        Reset(ref beforeIdentifierPoint);
-
-                        //clear fields that were populated after the reset point
-                        identifierOrThisOpt = null;
-                        typeParameterListOpt = null;
+                        /*
+                         * It feels like this recovery is applied too aggressively. For example, if there is no line break between the last
```

Comment:
>* It feels like this recovery is applied too aggressively. For example, if there is no line break between the last [](start = 25, length = 114)

This looks like a verbatim copy of a comment from a pull request. Please adjust it to simply reflect the rational behind the logic or explain the final implementation. For example, leaving "It feels like this recovery is applied too aggressively" statement is likely to confuse the reader. Etc.
Also, consider merging (concatenating) this comment with the comment above.